packages/apollo-gateway/src/__tests__/integration/networkRequests.test.ts

```diff
@@ -94,6 +97,59 @@ it('Extracts service definitions from remote storage', async () => {
   expect(gateway.schema!.getType('User')!.description).toBe('This is my User');
 });
 
+it.each([
+  ['warned', 'present'],
+  ['not warned', 'absent'],
+])('conflicting configurations are %s about when %s', async (_word, mode) => {
+  const isConflict = mode === 'present';
+  const spyConsoleWarn = jest.spyOn(console, 'warn');
+  let blockerResolve: () => void;
+  const blocker = new Promise(resolve => (blockerResolve = resolve));
+  const original = loadServicesFromStorage.getServiceDefinitionsFromStorage;
+  const spyGetServiceDefinitionsFromStorage = jest
+    .spyOn(loadServicesFromStorage, 'getServiceDefinitionsFromStorage')
```

Comment:
This seems like a responsible setting.  I somehow would be surprised if we're not leveraging it right now, but it would be worth investigating.  Will take a peak!

However, I do think we're already restoring this mock, [right before `spyConsoleWarn.mockRestore()` below](https://github.com/apollographql/apollo-server/pull/3868/files#diff-e627fbdfe08d54183ebc4f2db267b06eR149)?

Reply:
🤔 I totally missed that with my eyes somehow. Disregard!